pkg/dataobj/logs_reader.go

```diff
@@ -109,33 +113,79 @@ func (r *LogsReader) Read(ctx context.Context, s []Record) (int, error) {
 		}
 	}
 
-	r.buf = slices.Grow(r.buf, len(s))
+	r.buf = slicegrow.GrowToCap(r.buf, len(s))
 	r.buf = r.buf[:len(s)]
 
+	// Fill the row buffer with empty values so they can re-use the memory we pass in.
+	for i := range r.buf {
+		r.buf[i].Values = slicegrow.GrowToCap(r.buf[i].Values, len(r.columns))
```

Comment:
Doesn't the reader code effectively already do this, providing some kind of lazy initialization?